File: server/src/models/RideRequest.js
const mongoose = require('../db');

const RideRequestSchema = new mongoose.Schema({
  origin: { type: String, required: true },
  destination: { type: String, required: true },
  time: { type: String, required: true },
  seatCount: { type: Number, required: true },
});

module.exports = mongoose.model('RideRequest', RideRequestSchema);

File: mobile/lib/models/ride_request.dart
import 'package:flutter/material.dart';

class RideRequest {
  final String origin;
  final String destination;
  final TimeOfDay time;
  final int? seatCount;

  RideRequest({
    required this.origin,
    required this.destination,
    required this.time,
    this.seatCount,
  });
}
